vir/src/common/identifier.rs

```diff
@@ -1,3 +1,18 @@
 pub trait WithIdentifier {
     fn get_identifier(&self) -> String;
 }
+
+// TODO: this will need to be some extensible name, which supports all of the
+// name operations that we might want to do, but is not a string.
+
+/// Temporary standing for a name, returned by `NameInterner::intern`.
+/// Can be exchanged for a real `String` name with `NameInterner::get_interned`,
+/// but only once all names have been interned.
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash, Serialize, Deserialize, PartialOrd, Ord)]
+pub struct NameHash(u64, u64);
```

Comment:
Why is this not just a wrapper around `u64`? The interner could have a counter that is incremented each time it needs to intern a new identifier.

Reply:
Also, why not wrapping around `DefId` or, alternatively,  the full unique `item_name` string? The latter would be more stable than a counter when the order of the interned elements change.